src/game/client/water/w3dwater.cpp

```diff
@@ -0,0 +1,2327 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Water Render Object
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "w3dwater.h"
+#include "assetmgr.h"
+#include "baseheightmap.h"
+#include "camera.h"
+#include "dx8wrapper.h"
+#include "gamelogic.h"
+#include "globaldata.h"
+#include "light.h"
+#include "matinfo.h"
+#include "mesh.h"
+#include "polygontrigger.h"
+#include "rinfo.h"
+#include "rtsutils.h"
+#include "shadermanager.h"
+#include "surfaceclass.h"
+#include "texture.h"
+#include "vertmaterial.h"
+#include "w3dpoly.h"
+#include "w3dscene.h"
+#include "w3dshroud.h"
+#include "w3dwatertracks.h"
+#include "water.h"
+#include "xfer.h"
+
+#ifdef BUILD_WITH_D3D8
+#include <d3dx8.h>
+#endif
+
+ShaderClass g_zFillAlphaShader(0x184BB);
+
+void Do_Sky_Box_Set(bool b)
+{
+    if (g_theWriteableGlobalData) {
+        g_theWriteableGlobalData->m_drawSkyBox = b;
+    }
+}
+
+WaterRenderObjClass::WaterRenderObjClass() :
+    m_settings{},
+    m_dx(0.0f),
+    m_dy(0.0f),
+    m_indexBuffer(nullptr),
+    m_waterTrackSystem(nullptr),
+    m_doWaterGrid(false),
+    m_meshVertexMaterialClass(nullptr),
+    m_meshLight(nullptr),
+    m_vertexMaterialClass(nullptr),
+    m_alphaClippingTexture(nullptr),
+    m_useCloudLayer(true),
+    m_waterType(WATER_TYPE_0_TRANSLUCENT),
+    m_tod(TIME_OF_DAY_AFTERNOON),
+    m_pReflectionTexture(nullptr),
+    m_skyBox(nullptr),
+    m_vertexBufferD3D(nullptr),
+    m_indexBufferD3D(nullptr),
+    m_vertexBufferD3DOffset(0),
+    m_dwWavePixelShader(0),
+    m_meshData(nullptr),
+    m_meshDataCount(0),
+    m_meshInMotion(false),
+    m_gridOrigin(0.0f, 0.0f),
+    m_gridDirectionX(1.0f, 0.0f),
+    m_gridDirectionY(1.0f, 0.0f),
+    m_gridCellSize(10.0f),
+    m_gridCellsX(128),
+    m_gridCellsY(128),
+    m_gridWidth(m_gridCellsX * m_gridCellSize),
+    m_gridHeight(m_gridCellsY * m_gridCellSize),
+    m_riverVOrigin(0.0f),

```

Comment:
missing m_pBumpTexture or compiler makes it?

Reply:
fixed.